bliss/simulator/prior.py

```diff
@@ -134,25 +133,9 @@ def sample_prior(self, batch_image_ids) -> TileCatalog:
             The remaining dimensions are variable-specific.
         """
         locs = self._sample_locs()
-        select_gal_flux_ratios = []
-        select_star_flux_ratios = []
-
-        for batch_image_id in batch_image_ids:
-            image_id = tuple(batch_image_id)
-            # get value if key exists, otherwise pick randomly
-            select_gal_flux_ratios.append(
-                self.gals_fluxes[image_id]
-                if image_id in self.gals_fluxes
-                else random.choice(list(self.gals_fluxes.values()))
-            )
-            select_star_flux_ratios.append(
-                self.stars_fluxes[image_id]
-                if image_id in self.stars_fluxes
-                else random.choice(list(self.stars_fluxes.values()))
-            )
-
-        galaxy_fluxes, galaxy_params = self._sample_galaxy_prior(select_gal_flux_ratios)
-        star_fluxes = self._sample_star_fluxes(select_star_flux_ratios)
+
+        galaxy_fluxes, galaxy_params = self._sample_galaxy_prior(self.gals_fluxes)
+        star_fluxes = self._sample_star_fluxes(self.stars_fluxes)
```

Comment:
I don't really like this because `self.gals_fluxes` and `self.stars_fluxes` aren't attributes in `ImagePrior`, they're only in the subclasses. That can easily break if a subclass decides to implement/store ratios in a different way.

Imo it would be cleaner to have an abstract `get_star_ratios` function with a `n_samples` parameter (same for galaxies). Then `_sample_galaxy_prior` and `_sample_star_fluxes` can just call those respective functions and get the ratios themselves instead of having to pass them in.